### PR TITLE
use new gdrive-datastore to check and fix gdrive access

### PR DESCRIPTION
**Pull request by ChenglimEar**
This addresses the problem where we could not test gdrive access from a branch because the branch name was being used as the subfolder where the redacted data was stored on Google Drive.  We needed to set the subfolder to always be `main` since that is where the released redacted data lives.

This should not be approved until the workflow for checking Google Drive access runs successfully.  That workflow depends on the `SERVICE_ACCOUNT_KEY_JSON` secret and the `GDRIVE_FOLDER` variable being set right.

### CODE REVIEW

**Comment by mikeubell** [comment]
<details>
<summary>Build diff from Commit 17963a5a84b7b0200b99fe9acdd5ed83d38ea912:</summary>

```diff
diff --git a/build/_data/stats.json b/build/_data/stats.json
index f0b04ca..d284f2d 100644
--- a/build/_data/stats.json
+++ b/build/_data/stats.json
@@ -1,3 +1,3 @@
 {
-  "date_processed": "2023-10-29 00:07:45 -0700"
+  "date_processed": "2023-10-30 00:07:25 -0700"
 }
```
</details>

**Comment by mikeubell** [comment]
<details>
<summary>Build diff from Commit 13c7e74fe9572a11a1ab1b0c76db2a11ff828a52:</summary>

```diff
diff --git a/build/_data/stats.json b/build/_data/stats.json
index f0b04ca..d284f2d 100644
--- a/build/_data/stats.json
+++ b/build/_data/stats.json
@@ -1,3 +1,3 @@
 {
-  "date_processed": "2023-10-29 00:07:45 -0700"
+  "date_processed": "2023-10-30 00:07:25 -0700"
 }
```
</details>

**Comment by mikeubell** [comment]
<details>
<summary>Build diff from Commit 84283ed3348dbcd96c97aeb8fe5f4e9417a9a89d:</summary>

```diff
diff --git a/build/_data/stats.json b/build/_data/stats.json
index f0b04ca..d284f2d 100644
--- a/build/_data/stats.json
+++ b/build/_data/stats.json
@@ -1,3 +1,3 @@
 {
-  "date_processed": "2023-10-29 00:07:45 -0700"
+  "date_processed": "2023-10-30 00:07:25 -0700"
 }
```
</details>

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,7 +12,6 @@ jobs:
       GDRIVE_FOLDER: ${{ vars.GDRIVE_FOLDER }}
     steps:
     - uses: actions/checkout@v3
-    - run: 'echo "hello"'
     - run: "pip install -r gdrive_requirements.txt"
```
Comment (ckingbailey) [inline .github/workflows/main.yml:15]: No quotes needed here. Sorry, I realize this is not specifically in the scope of these changes, but I noticed it.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,7 +12,6 @@ jobs:
...
     - run: "python test_pull_from_gdrive.py"
```
Comment (ckingbailey) [inline .github/workflows/main.yml:16]: Same as above, quotes unnecessary

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,8 +1,9 @@
+#name: "Generate Website Data"
```
Comment (ckingbailey) [inline .github/workflows/main.yml:1]: Rm commented-out line

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -11,7 +12,6 @@ jobs:
       GDRIVE_FOLDER: ${{ vars.GDRIVE_FOLDER }}
     steps:
     - uses: actions/checkout@v3
-    - run: 'echo "hello"'
     - run: "pip install -r gdrive_requirements.txt"
     - run: "python test_pull_from_gdrive.py"
     - name: Archive pulled files
```
Comment (ckingbailey) [inline .github/workflows/main.yml:17]: GH interface won't let me comment on the line below. upload-artifact should be bumped to v3 there. v2 is resulting in a "Node12 is deprecated warning".

File: .github/workflows/verify-gdrive.yml
Comment (ckingbailey) [inline .github/workflows/verify-gdrive.yml:1]: Same comments as on other workflow: unnecessary quotes, bump upload-artifact

File: .devcontainer/postCreateCommand.sh
Comment (ckingbailey) [inline .devcontainer/postCreateCommand.sh:1]: I don't think we should use camelCase for shell scripts. I've always seen them kebab-case or snake_case

**Review by ckingbailey — CHANGES_REQUESTED** [review]
Mostly just some nit-picky items, and bumping of a package version

File: .github/workflows/main.yml (hunk 1 above)
Comment (ChenglimEar) [reply .github/workflows/main.yml]: removed quotes

File: .github/workflows/main.yml (hunk 2 above)
Comment (ChenglimEar) [reply .github/workflows/main.yml]: removed quotes

File: .github/workflows/main.yml (hunk 3 above)
Comment (ChenglimEar) [reply .github/workflows/main.yml]: Updated the comment to explain what's going on.  This is a transitional state.  After merging into the master branch, GitHub will be aware of two workflows and then I can make the required changes on a branch to turn this workflow into something else.

File: .github/workflows/main.yml (hunk 4 above)
Comment (ChenglimEar) [reply .github/workflows/main.yml:20]: changed to v3

File: .github/workflows/verify-gdrive.yml
Comment (ChenglimEar) [reply .github/workflows/verify-gdrive.yml:1]: changed

**Comment by mikeubell** [comment]
<details>
<summary>Build diff from Commit 93f7f56e947377c74a9dd49fddecf7f25f0e91a6:</summary>

```diff
diff --git a/build/_data/stats.json b/build/_data/stats.json
index f0b04ca..d284f2d 100644
--- a/build/_data/stats.json
+++ b/build/_data/stats.json
@@ -1,3 +1,3 @@
 {
-  "date_processed": "2023-10-29 00:07:45 -0700"
+  "date_processed": "2023-10-30 00:07:25 -0700"
 }
```
</details>

File: .devcontainer/postCreateCommand.sh
Comment (ChenglimEar) [reply .devcontainer/postCreateCommand.sh:1]: changed

File: .github/workflows/verify-gdrive.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name: "New Check Google Drive Access"
+on:
+  workflow_dispatch:
+jobs:
+  check:
+    runs-on: ubuntu-latest
+    env:
+      REPO_OWNER: ${{ github.repository_owner}}
+      REPO_BRANCH: ${{ github.ref_name }}
+      SERVICE_ACCOUNT_KEY_JSON: ${{ secrets.SERVICE_ACCOUNT_KEY_JSON }}
```
Comment (tdooner) [inline .github/workflows/verify-gdrive.yml:10]: This SERVICE_ACCOUNT_KEY_JSON... this was what you downloaded directly from the Google Developers Console - right?

**Review by tdooner — APPROVED** [review]

**Comment by tdooner** [comment]
(I don't really have much context on this anymore, though, so you should probably take my review with a grain of salt.)

**Comment by mikeubell** [comment]
<details>
<summary>Build diff from Commit f4f219bea0a166cf98f79c9eb5e426ee88ec17ec:</summary>

```diff
diff --git a/build/_data/committees/1410941.json b/build/_data/committees/1410941.json
index 178a733..fcb85cd 100644
--- a/build/_data/committees/1410941.json
+++ b/build/_data/committees/1410941.json
@@ -153,7 +153,7 @@
       "Tran_Date": "2018-09-24",
       "Tran_NamF": null,
       "Tran_NamL": "Morse Management, Inc.",
-      "Tran_Zip4": "10022"
+      "Tran_Zip4": "94661"
     },
     {
       "Filer_ID": "1410941",
@@ -164,7 +164,7 @@
       "Tran_Date": "2018-09-24",
       "Tran_NamF": null,
       "Tran_NamL": "Morse Management, Inc.",
-      "Tran_Zip4": "94661"
+      "Tran_Zip4": "10022"
     },
     {
       "Filer_ID": "1410941",
@@ -813,7 +813,7 @@
       "Tran_Date": "2018-10-19",
       "Tran_NamF": "Garrett",
       "Tran_NamL": "Riegg",
-      "Tran_Zip4": "94602"
+      "Tran_Zip4": "94606"
     },
     {
       "Filer_ID": "1410941",
@@ -824,7 +824,7 @@
       "Tran_Date": "2018-10-19",
       "Tran_NamF": "Garrett",
       "Tran_NamL": "Riegg",
-      "Tran_Zip4": "94606"
+      "Tran_Zip4": "94602"
     },
     {
       "Filer_ID": "1410941",
diff --git a/build/_data/committees/1421001.json b/build/_data/committees/1421001.json
index 09cc4a0..ff4043a 100644
--- a/build/_data/committees/1421001.json
+++ b/build/_data/committees/1421001.json
@@ -1198,7 +1198,7 @@
       "Tran_Date": "2020-07-15",
       "Tran_NamF": "Jonathan",
       "Tran_NamL": "Williams",
-      "Tran_Zip4": "94602"
+      "Tran_Zip4": "94603"
     },
     {
       "Filer_ID": "1421001",
@@ -1209,7 +1209,7 @@
       "Tran_Date": "2020-07-15",
       "Tran_NamF": "Jonathan",
       "Tran_NamL": "Williams",
-      "Tran_Zip4": "94603"
+      "Tran_Zip4": "94602"
     },
     {
       "Filer_ID": "1421001",
diff --git a/build/_data/elections/oakland/2018-06-05.json b/build/_data/elections/oakland/2018-06-05.json
index b39b338..140316d 100644
--- a/build/_data/elections/oakland/2018-06-05.json
+++ b/build/_data/elections/oakland/2018-06-05.json
@@ -57,7 +57,7 @@
       "total_contributions": 15000.0
     },
     {
-      "name": "Service Employees International Union Local 1021 Issues PAC",
+      "name": "Oakland Athletics Baseball Company",
       "type": "Measure",
       "election_name": "oakland-june-2018",
       "total_contributions": 10000.0
diff --git a/build/_data/elections/oakland/2022-11-08.json b/build/_data/elections/oakland/2022-11-08.json
index f716008..d436f62 100644
--- a/build/_data/elections/oakland/2022-11-08.json
+++ b/build/_data/elections/oakland/2022-11-08.json
@@ -2,13 +2,13 @@
   "total_contributions": 6476065.89,
   "total_contributions_by_source": {
     "Out of State": 521147.35,
-    "Within Oakland": 2632398.1800000006,
+    "Within Oakland": 2632398.1799999997,
     "Within California": 3006531.1199999996
   },
   "contributions_by_type": {
     "PTY": 15900.0,
     "Committee": 995101.14,
-    "Individual": 2507910.870000002,
+    "Individual": 2507910.869999998,
     "Unitemized": 135828.3,
     "Self Funding": 2711.0,
     "Other (includes Businesses)": 2638453.6399999997
@@ -107,7 +107,7 @@
       "total_contributions": 6300.0
     },
     {
-      "name": "Russ Taplin",
+      "name": "John Protopappas",
       "type": "Office",
       "election_name": "oakland-2022",
       "total_contributions": 5400.0
diff --git a/build/_data/elections/oakland/2023-11-07.json b/build/_data/elections/oakland/2023-11-07.json
index 73ff78f..1d8179a 100644
--- a/build/_data/elections/oakland/2023-11-07.json
+++ b/build/_data/elections/oakland/2023-11-07.json
@@ -64,19 +64,19 @@
   ],
   "top_contributors_for_offices": [
     {
-      "name": "UA Local 342",
+      "name": "Service Employees International Union Local 1021 Candidate PAC",
       "type": "Office",
       "election_name": "oakland-2023",
       "total_contributions": 1200.0
     },
     {
-      "name": "Service Employees International Union Local 1021 Candidate PAC",
+      "name": "Laborers Local 304",
       "type": "Office",
       "election_name": "oakland-2023",
       "total_contributions": 1200.0
     },
     {
-      "name": "Laborers Local 304",
+      "name": "UA Local 342",
       "type": "Office",
       "election_name": "oakland-2023",
       "total_contributions": 1200.0
diff --git a/build/_data/elections/oakland/2024-11-05.json b/build/_data/elections/oakland/2024-11-05.json
index 129d546..adbcd09 100644
--- a/build/_data/elections/oakland/2024-11-05.json
+++ b/build/_data/elections/oakland/2024-11-05.json
@@ -49,19 +49,19 @@
   ],
   "top_contributors_for_offices": [
     {
-      "name": "Weylin White",
+      "name": "Todd Scanlin",
       "type": "Office",
       "election_name": "oakland-2024",
       "total_contributions": 1800.0
     },
     {
-      "name": "Todd Scanlin",
+      "name": "Dow Terry",
       "type": "Office",
       "election_name": "oakland-2024",
       "total_contributions": 1800.0
     },
     {
-      "name": "Jacob Zonn",
+      "name": "Brooke Levin",
       "type": "Office",
       "election_name": "oakland-2024",
       "total_contributions": 1800.0
diff --git a/build/_data/stats.json b/build/_data/stats.json
index f0b04ca..d284f2d 100644
--- a/build/_data/stats.json
+++ b/build/_data/stats.json
@@ -1,3 +1,3 @@
 {
-  "date_processed": "2023-10-29 00:07:45 -0700"
+  "date_processed": "2023-10-30 00:07:25 -0700"
 }
diff --git a/build/_data/totals.json b/build/_data/totals.json
index 4d36b47..83bf617 100644
--- a/build/_data/totals.json
+++ b/build/_data/totals.json
@@ -782,13 +782,13 @@
     "total_contributions": 6476065.89,
     "total_contributions_by_source": {
       "Out of State": 521147.35,
-      "Within Oakland": 2632398.1800000006,
+      "Within Oakland": 2632398.1799999997,
       "Within California": 3006531.1199999996
     },
     "contributions_by_type": {
       "PTY": 15900.0,
       "Committee": 995101.14,
-      "Individual": 2507910.870000002,
+      "Individual": 2507910.869999998,
       "Unitemized": 135828.3,
       "Self Funding": 2711.0,
       "Other (includes Businesses)": 2638453.6399999997
@@ -887,7 +887,7 @@
         "total_contributions": 6300.0
       },
       {
-        "name": "Russ Taplin",
+        "name": "John Protopappas",
         "type": "Office",
         "election_name": "oakland-2022",
         "total_contributions": 5400.0
@@ -1033,7 +1033,7 @@
         "total_contributions": 15000.0
       },
       {
-        "name": "Service Employees International Union Local 1021 Issues PAC",
+        "name": "Oakland Athletics Baseball Company",
         "type": "Measure",
         "election_name": "oakland-june-2018",
         "total_contributions": 10000.0
@@ -1377,19 +1377,19 @@
     ],
     "top_contributors_for_offices": [
       {
-        "name": "UA Local 342",
+        "name": "Service Employees International Union Local 1021 Candidate PAC",
         "type": "Office",
         "election_name": "oakland-2023",
         "total_contributions": 1200.0
       },
       {
-        "name": "Service Employees International Union Local 1021 Candidate PAC",
+        "name": "Laborers Local 304",
         "type": "Office",
         "election_name": "oakland-2023",
         "total_contributions": 1200.0
       },
       {
-        "name": "Laborers Local 304",
+        "name": "UA Local 342",
         "type": "Office",
         "election_name": "oakland-2023",
         "total_contributions": 1200.0
@@ -1493,19 +1493,19 @@
     ],
     "top_contributors_for_offices": [
       {
-        "name": "Weylin White",
+        "name": "Todd Scanlin",
         "type": "Office",
         "election_name": "oakland-2024",
         "total_contributions": 1800.0
       },
       {
-        "name": "Todd Scanlin",
+        "name": "Dow Terry",
         "type": "Office",
         "election_name": "oakland-2024",
         "total_contributions": 1800.0
       },
       {
-        "name": "Jacob Zonn",
+        "name": "Brooke Levin",
         "type": "Office",
         "election_name": "oakland-2024",
         "total_contributions": 1800.0
```
</details>

File: .github/workflows/verify-gdrive.yml (hunk 5 above)
Comment (ChenglimEar) [reply .github/workflows/verify-gdrive.yml:10]: yes.  It is downloaded as a JSON file - the key for one of the service accounts.  The contents of that file is placed in the SERVICE_ACCOUNT_KEY_JSON secret and passed down to python code where it is used to access Google Drive using the credentials of the service account.

**Review by ckingbailey — APPROVED** [review]

**Pull request merged**